ruta-core/src/main/java/org/apache/uima/ruta/rule/RegExpRule.java

```diff
@@ -85,7 +86,7 @@ public ScriptApply apply(RutaStream stream, InferenceCrowd crowd) {
     Map<Integer, List<Type>> groupTypes = getGroup2Types(context, stream);
     Map<Integer, Map<Type, Map<String, Object>>> fa = getFeatureAssignmentMap(stream);
 
-    Pattern pattern = Pattern.compile(regexpString, Pattern.MULTILINE | Pattern.DOTALL);
+    Pattern pattern = RutaPatternCache.getPattern(regexpString, false);

```

Comment:
How about making the `getPattern` API more like the `compile` API by having the `MULTILINE | DOTALL` here instead of the `false`? That would make the code more readable (what's false anyway?) and more versatile.